includes/tax-functions.php

```diff
@@ -442,3 +410,63 @@ function edd_download_is_tax_exclusive( $download_id = 0 ) {
 
 	return (Bool) apply_filters( 'edd_download_is_tax_exclusive', $ret, $download_id );
 }
+
+/**
+ * Gets the tax rate object from the database for a given country / region.
+ * Used in `edd_get_tax_rate`, `edd_build_order`, `edd_add_manual_order`.
+ *
+ * @param array $args
+ * @return object|boolean
+ *
+ * @since 3.0
+ */
+function edd_get_tax_rate_object( $args ) {
+
+	$rate = false;
+	if ( empty( $args['country'] ) ) {
+		return $rate;
+	}
+
+	// Fetch all the tax rates from the database.
+	// The region is not passed in deliberately in order to check for country-wide tax rates.
+	$tax_rates = edd_get_tax_rates(
+		array(
+			'name'   => $args['country'],
+			'status' => 'active',
+		),
+		OBJECT
+	);
+
+	// Save processing if only one tax rate is returned.
+	if ( 1 === count( $tax_rates ) ) {
+		$tax_rate = $tax_rates[0];
+
+		if ( $tax_rate->name === $args['country'] && $tax_rate->description === $args['region'] ) {
+			return $tax_rate;
+		}
+	}
+
+	if ( ! empty( $tax_rates ) ) {
+		foreach ( $tax_rates as $tax_rate ) {
+
+			// Countrywide tax rate.
+			if ( 'country' === $tax_rate->scope ) {
+				$rate = $tax_rate;
+
+			// Regional tax rate.
+			} else {
+				if ( empty( $tax_rate->description ) || strtolower( $args['region'] ) !== strtolower( $tax_rate->description ) ) {
+					continue;
+				}
+
+				$regional_rate = $tax_rate->amount;
+
+				if ( ( 0 !== $regional_rate || ! empty( $regional_rate ) ) && '' !== $regional_rate ) {
+					$rate = $tax_rate;
+				}
+			}
+		}
```

Comment:
@ashleyfae as it stands, if there are multiple tax rates retrieved, the last one is what will be chosen--even if `$rate` is defined in a prior loop, it can be overwritten by a later one. Is that the correct behavior, or should we tuck something in to go ahead and return the first matched object?

Reply:
Two things to consider I think:

1. You shouldn't be able to have multiple active country-wide tax rates for the same country. (Or if we haven't fixed that yet then we will.. I recall an issue on it at one point.) So hopefully we shouldn't run into that issue.
2. If you have a country-wide tax rate **and** a tax rate for the same country that's narrowed down to a region, then I think the region one should always win. I don't think that's currently happening. However, maybe that's a big/separate enough change that we should make a new issue for it.

Reply:
The multiple country-wide tax rates are still an issue, yes.

So I'm reviewing the logic in this foreach loop and it's currently a bit convoluted, and I think could stand to be simplified ( `0`, `''`, both evaluate the same as `empty` in the regional check). Going to work on that and see if I can finesse it.

Reply:
Update: have reworked the logic so that if a regional rate is set, it will automatically be returned and the `foreach` stopped. If a country rate is set, it will be stored, but can be overridden in the next iteration of the loop, I think.

@JJJ per your question above, a regional rate should take precedence over a country rate.